### ISSUE TITLE
GetInitiatorIqns does not ignore comments

### ISSUE THREAD

**Issue by rjbrown57** [NetApp/trident#681]
GetInitiatorIqns does not ignore text in format #InitiatorName=
(https://github.com/NetApp/trident/blob/16e84e6826b6591cb33aae9ccdb120918d3c0eae/utils/osutils.go#L334)

- Trident version: trident:21.07.1

**To Reproduce**
Create a /etc/iscsi/initiatorname.iscsi with contents #InitiatorName=anythinghere

**Expected behavior**
Ignore comments

**Comment by balaramesh** [comment]
@rjbrown57 I gave this a try. Here's my /etc/iscsi/initiatorname.iscsi file:

```
#InitiatorName=iqn.1994-05.demo.netapp.com
#blankline
#anotherblankline
```

Upon restarting `iscisd` and `iscsi`, I found this in the logs on the daemonset pod for the node in question:

```
time="2021-12-08T18:39:03Z" level=debug msg=">>>> osutils.GetInitiatorIqns" requestID=12b23cfd-3b93-4d5d-a44f-3f13751282b8 requestSource=Internal
time="2021-12-08T18:39:03Z" level=debug msg=">>>> osutils.execCommand." args="[/etc/iscsi/initiatorname.iscsi]" command=cat requestID=12b23cfd-3b93-4d5d-a44f-3f13751282b8 requestSource=Internal
time="2021-12-08T18:39:03Z" level=info msg="Starting periodic node access reconxciliation service." requestID=50bf774b-d9c7-4e94-b9fe-40267a4650da requestSource=Periodic                                                                     
time="2021-12-08T18:39:03Z" level=debug msg="<<<< osutils.execCommand." command=cat error="<nil>" output="#InitiatorName=iqn.1994-05.demo.netapp.com\n#blankline\n#anotherblankline" requestID=12b23cfd3b93-4d5d-a44f-3f13751282b8 requestSource=Internal                                                                  x
time="2021-12-08T18:39:03Z" level=debug msg="<<<< osutils.GetInitiatorIqns" requestID=12b23cfd-3b93-4d5d-a44f-3f13751282b8 requestSource=Internal
time="2021-12-08T18:39:03Z" level=info msg="Discovered iSCSI initiator name." IQN=iqn.1994-05.demo.netapp.comrequestID=12b23cfd-3b93-4d5d-a44f-3f13751282b8 requestSource=Internal 
```

Trident discovers the IQN incorrectly when `/etc/iscsi/initiatorname.iscsi` contains a commented line with InitiatorName.

While this is fixed in Trident, the best way to avoid this would be to ensure `/etc/iscsi/initiatorname.iscsi` contains a single InitiatorName entry

**Comment by rjbrown57** [comment]
@balaramesh I opened this MR with a fix.

**Comment by gnarl** [comment]
This issue is fixed with [commit 43372f1](https://github.com/NetApp/trident/commit/43372f1b422b4b0b0743f99fc1fbaa9ddb3efab3) and will be included in the Trident v22.04 release.
